registry/consul/utils.go

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package consul
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+)
+
+import (
+	consul "github.com/hashicorp/consul/api"
+	perrors "github.com/pkg/errors"
+)
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/utils"
+)
+
+func buildId(url common.URL) string {
+	t := md5.Sum([]byte(url.String()))
+	return hex.EncodeToString(t[:])
+}
+
+func buildService(url common.URL) (*consul.AgentServiceRegistration, error) {
+	var err error
+
+	// id
+	id := buildId(url)
+
+	// address
+	if url.Ip == "" {
+		url.Ip, _ = utils.GetLocalIP()
+	}
+
+	// port
+	port, err := strconv.Atoi(url.Port)
+	if err != nil {
+		return nil, err
+	}
+
+	// tcp
+	tcp := fmt.Sprintf("%s:%d", url.Ip, port)
+
+	// tags
+	tags := make([]string, 0)
+	for k := range url.Params {
+		tags = append(tags, k+"="+url.Params.Get(k))
+	}
+	tags = append(tags, "dubbo")
+
+	// meta
+	meta := make(map[string]string)
```

Comment:
give it a len param.

Reply:
I have changed the default cap of `list` and `map` to `8`.